File: conection/lib/models/motel.dart
import 'image.dart';
import 'detail.dart';
class Motel{
  int id;
  String typeLive;
  String title;
  int price;
  String priceType;
  DateTime dateUpdate;
  DateTime dateDue;
  String time;
  String status;
  bool verify;
  String address;
  String description;
  String phone;
  String typemotel;
  int areaZone;
  String areaZoneType;
  String typeservice;
  String city;
  int cityId;
  String province;
  int provinceId;
  String district;
  int districtId;
  String street;
  int streetId;
  List<image> images;
  String bills;
  String user;
  int userId;
  detail Detail;


  Motel(int id,String typeLive,String title,
      int price, String priceType, DateTime dateUpdate,
      DateTime dateDue, String time, String status,
      bool verify, String address, String description,
      String phone, String typemotel, int areaZone,
      String areaZoneType, String typeservice, String city,
      int cityId, String province, int provinceId,
      String district, int districtId, String street
      ,int streetId, List<image> images, String bills,String user,
      int userId, detail Detail)
  {
    this.id=id;
    this.typeLive=typeLive;
    this.title=title;
    this.price = price;
    this.priceType = priceType;
    this.dateUpdate= dateUpdate;
    this.dateDue= dateDue;
    this.time= time;
    this.status= status;
    this.verify= verify;
    this.address = address;
    this.description= description;
    this.phone= phone;
    this.typemotel= typemotel;
    this.areaZone = areaZone;
    this.areaZoneType= areaZoneType;
    this.typeservice = typeservice;
    this.city= city;
    this.cityId= cityId;
    this.province= province;
    this.provinceId= provinceId;
    this.district=district;
    this.districtId=districtId;
    this.street= street;
    this.streetId=streetId;
    this.images= images;
    this.bills=bills;
    this.user=user;
    this.userId=userId;
    this.Detail=Detail;

  }
  Motel.fromJson(Map json):id=json['id'],
  typeLive=json['typeLive'],
  title=json['title'],
  price = json['price'],
  priceType = json['priceType'],
  dateUpdate= json['dateUpdate'],
  dateDue= json['dateDue'],
  time= json['time'],
  status= json['status'],
  verify= json['verify'],
  address = json['address'],
  description= json['description'],
  phone= json['phone'],
  typemotel= json['typemotel'],
  areaZone = json['areaZone'],
  areaZoneType= json['areaZoneType'],
  typeservice = json['typeservice'],
  city= json['city'],
  cityId= json['cityId'],
  province= json['province'],
  provinceId= json['provinceId'],
  district=json['district'],
  districtId=json['districtId'],
  street= json['street'],
  streetId=json['streetId'],
  images= json['images'],
  bills=json['bills'],
  user=json['user'],
  userId=json['userId'],
  Detail=json['Detail'];
}

File: Sourcode/back/Websitedangtintimkiemnhatro/Websitedangtintimkiemnhatro/Models/Motel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Websitedangtintimkiemnhatro.Models
{
    public class Motel
    {
        [Key]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Price { get; set; }
        public string PriceType { get; set; }
        public DateTime DateUpdate { get; set; }
        public DateTime DateDue { get; set; }
        public string Time { get; set; } //Chọn loại thanh toán ngày, tháng ,tuần
        public string Status { get; set; }
        public bool Verify { get; set; }
        public string Address { get; set; }
        public string AddressNumber { get; set; }
        public string Description { get; set; }
        public string Phone { get; set; }
        public string Typemotel { get; set; } // Mua bán hay thuê
        public string AreaZone { get; set; }
        public string AreaZoneType { get; set; }
        public string Typeservice { get; set; }
        public string Latitude { get; set; }
        public string Longitude { get; set; }
        public City City { get; set; }
        [ForeignKey("CityId")]
        [Required]
        public int CityId { get; set; }
        public Province Province { get; set; }
        [ForeignKey("ProvinceId")]
        [Required]
        public int ProvinceId { get; set; }

        public District District { get; set; }
        [ForeignKey("DistrictId")]
        [Required]
        public int DistrictId { get; set; }
        public Street Street { get; set; }
        [ForeignKey("StreetId")]
        [Required]
        public int StreetId { get; set; }

        public ICollection<Image> Images { get; set; }
        public ICollection<Bill> Bills { get; set; }
        public User User { get; set; }
        [ForeignKey("UserId")]
        [Required]
        public int UserId { get; set; }
        public Detail Detail { get; set; }

    }
}
